lib/report/report.go

```diff
@@ -42,12 +43,13 @@ func Collect(ctx context.Context, config Config, w io.Writer) error {
 	for _, filter := range teleutils.Deduplicate(config.Filters) {
 		switch filter {
 		case FilterSystem:
-			collectors = append(collectors, NewSystemCollector()...)
+			collectors = append(collectors, NewSystemCollector(config.Since)...)
 			collectors = append(collectors, NewPackageCollector(config.Packages))
 		case FilterKubernetes:
-			collectors = append(collectors, NewKubernetesCollector(ctx, utils.Runner)...)
+			collectors = append(collectors, NewKubernetesCollector(ctx, utils.Runner, config.Since)...)
 		case FilterEtcd:
 			collectors = append(collectors, etcdBackup()...)
+			collectors = append(collectors, etcdMetrics()...)
```

Comment:
I'm not familiar enough with the filters, but based on your sample output, this seems to only collect metrics from one node in the cluster. Some of the metrics will be relevant to the particular server though, so this should likely be collected on each master node, so each server can be inspected individually. Be careful not to collect on the worker nodes, as those connections will be randomly load balanced to a master.

Unfortunately what makes this even more complicated, is many of the metrics are not useful on their own with a single snapshot. Many metrics will be counters since startup, which don't say anything on their own. It's only when you delta the current snapshot, with one a minute ago, to you get a picture of what was happening over the last minute. This isn't really just isolated to etcd, I don't think we've really planned on how to pull any metric histories as part of the debug reporting. So we might want to put together a plan on pulling out valuable metrics histories.

Reply:
Updated so that etcd metrics are collected on all master nodes fab170f.

We should definitely look to do some additional processing of the metrics. For this PR I think the single snapshot can still provide some useful information?